core/types/json.go

```diff
@@ -44,6 +51,13 @@ func (b *hexBytes) UnmarshalJSON(input []byte) error {
 
 type hexBig big.Int
 
+func (b *hexBig) MarshalJSON() ([]byte, error) {
```

Comment:
I don't see how "0X" is valid. It's stupid enough that all of our API need to accept "0x", which is also not 'valid hex'. The RPC API documentation does not mention "0X" and I don't know any software library that would ever encode hex with "0X" as prefix.